compiler-plugin/src/main/kotlin/com/google/devtools/ksp/symbol/impl/utils.kt

```diff
@@ -484,3 +488,140 @@ private val PropertyDescriptor.declaresDefaultValue: Boolean
         is KtParameter -> declaration.defaultValue != null
         else -> false
     }
+
+fun <T : Annotation> KSAnnotated.getAnnotation(annotationKClass: KClass<T>): T? {
+    return this.annotations.firstOrNull {
+        annotationKClass.qualifiedName == it.annotationType.resolve().declaration.qualifiedName?.asString()
+    }?.toAnnotation(annotationKClass)
+}
+
+fun KSAnnotated.getAnnotations(): Sequence<out Annotation> {
+    return this.annotations.map {
+        val kClass = Class.forName(it.annotationType.resolve().declaration.qualifiedName?.asString()).kotlin
+        it.toAnnotation(kClass as KClass<Annotation>)
+    }
+}
+
+fun <T : Annotation> KSAnnotated.getAnnotationsByType(annotationKClass: KClass<T>): Sequence<T> {
+    return this.annotations.filter {
+        it.annotationType.resolve().declaration.qualifiedName?.asString() == annotationKClass.qualifiedName
+    }.map { it.toAnnotation(annotationKClass) }
+}
+
+fun <T : Annotation> KSAnnotated.isAnnotationPresent(annotationKClass: KClass<T>): Boolean =
+    getAnnotation(annotationKClass)?.let { true } ?: false
```

Comment:
this can be `getAnnotation(annotationKClass) != null`

Reply:
good call